src/Symfony/Component/Serializer/Encoder/JsonDecode.php

```diff
@@ -95,8 +95,11 @@ public function decode($data, $format, array $context = array())
             $decodedData = json_decode($data, $associative, $recursionDepth);
         }
 
-        if (JSON_ERROR_NONE !== $this->lastError = json_last_error()) {
-            throw new UnexpectedValueException(JsonEncoder::getLastErrorMessage());
+        if (false === $decodedData) {
```

Comment:
`$decodedData` can be false without it being an error if you decode the `"false"` string.

Reply:
No, json_decode('false') return string 'false', not boolean false.  false !== 'false'

Reply:
@diversantvlz you should try running it :)

Reply:
Ok, like this?